.pre-commit-config.yaml

```diff
@@ -32,6 +32,7 @@ repos:
       - nox
       - packaging
       - pygithub
+      - pytest
```

Comment:
Does having so many non pinned dependencies do not make pre-commit mypy fragile?

Reply:
A little (though pytest isn't too bad). Though generally if something changes we just adapt to the change.

Pre-commit doesn't have a way to pin these. Maybe eventually uv will.

Reply:
> Pre-commit doesn't have a way to pin these. Maybe eventually uv will.

What did you mean by `uv` will? pre-commit.ci do not allow to pass any constraints. 

The only possible way that I see is to have own CI job to bump constraints.

Reply:
I mean if uv adds multiple independent locked environments (and tasks), we could use that to move this check to uv.

Reply:
in napari (https://github.com/napari/napari) we use uv to compile a constraints file and call it from tox. 

Maybe the same approach could be used here?

Reply:
If this becomes too much of a problem, we could probably do something (setting it up with nox would be easy). But for now it's fine, and long term we might end up moving from mypy to red-knot anyway.